homeassistant/components/kodi/binary_sensor.py

```diff
@@ -0,0 +1,115 @@
+"""Sensor classes for Kodi."""
+from __future__ import annotations
+
+import logging
+
+from jsonrpc_base.jsonrpc import TransportError
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_CONNECTION, DOMAIN
+from .kodi_connman import KodiConnectionClient, KodiConnectionManager
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Kodi sensors based on a config entry."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    connman = data[DATA_CONNECTION]
+    if (uid := config_entry.unique_id) is None:
+        uid = config_entry.entry_id
+
+    sensor_entities = [
+        KodiBinaryEntity(
+            "Screensaver",
+            "GUI.OnScreensaverActivated",
+            "GUI.OnScreensaverDeactivated",
+            "System.ScreenSaverActive",
+            uid,
+            connman,
+        ),
+        KodiBinaryEntity(
+            "Energy saving",
+            "GUI.OnDPMSActivated",
+            "GUI.OnDPMSDeactivated",
+            "System.DPMSActive",
+            uid,
+            connman,
+        ),
+    ]
+    async_add_entities(sensor_entities)
+
+
+class KodiBinaryEntity(KodiConnectionClient, BinarySensorEntity):
+    """Generic binary sensor entity for WebSocket callbacks."""
+
+    def __init__(
+        self,
+        name: str,
+        api_on: str,
+        api_off: str,
+        boolean_name: str,
+        uid: str,
+        connman: KodiConnectionManager,
+    ) -> None:
+        """Initialize kodi binary sensor."""
+        # Define used websocket methods including their callbacks
+        super().__init__(connman, {api_on: self.async_on, api_off: self.async_off})
+
+        self._attr_is_on = False
+        self._attr_name = name
+        self._attr_unique_id = f"{uid}_{api_on}"
+        self._kodi_boolean = boolean_name
+
+    @callback
+    def async_on(self, sender, data):  # pylint: disable=unused-argument
```

Comment:
Can we add type annotations on these callbacks?

Reply:
I would, but I can't figure out the type of those (sender and data). Since we don't use them anyway would `Any` be OK? Like I did for `__init__` on the new base class.

Reply:
I think `Any` would be ok here, along with underscores in the name